x-pack/legacy/plugins/infra/public/containers/logs/log_analysis/api/ml_setup_module_api.ts

```diff
@@ -12,6 +12,21 @@ import { throwErrors, createPlainError } from '../../../../../common/runtime_typ
 
 const MODULE_ID = 'logs_ui_analysis';
 
+// This is needed due to: https://github.com/elastic/kibana/issues/43671
+const removeSampleDataIndex = (indexPattern: string) => {
+  const SAMPLE_DATA_INDEX = 'kibana_sample_data_logs*';
+  const indices = indexPattern.split(',');
+  const sampleDataIndex = indices.findIndex((index: string) => {
+    return index === SAMPLE_DATA_INDEX;
+  });
+  if (sampleDataIndex > -1) {
+    indices.splice(sampleDataIndex, 1);
+    return indices.join(',');
+  } else {
+    return indexPattern;
+  }
+};
```

Comment:
Would this work with a `filter()` too? If so that might be easier to follow later when we want to remember what this does and safely rip it out ... maybe it's just me but I always forget how indices and `-1` and Array#splice all work :P
```
return indices.filter(index => index !== SAMPLE_DATA_INDEX).join(',')
```

Reply:
👍 I'll change this on the "cleanup / bug fixing" PR so I don't need to trigger another build cycle.